mobile/lib/modules/album/services/album.service.dart

```diff
@@ -170,6 +170,32 @@ class AlbumService {
     return changes;
   }
 
+  /// Checks remote album for changes,
+  /// updates the local database and returns `true` if there were any changes
+  Future<bool> refreshRemoteAlbum({
+    required AlbumResponseDto remote,
+    required Album album,
+  }) async {
+    if (!_remoteCompleter.isCompleted) {
+      // guard against concurrent calls
+      return _remoteCompleter.future;
```

Comment:
this is the same completer used for "all remote album refresh": if this new single album refresh is running and a full request is waiting, it will at some point resolve without actually refreshing all albums.

Reply:
Thinking some more about it: we can remove the completer check in this method.

The sync service still has the lock. So running both all album refresh and an individual album refresh should be ok.